src/main/java/spoon/support/compiler/VirtualFolder.java

```diff
@@ -45,15 +45,17 @@ public void addFolder(SpoonFolder o) {
 
 	@Override
 	public List<SpoonFile> getAllFiles() {
-		List<SpoonFile> result = new ArrayList<>();
+		Set<SpoonFile> result = new HashSet<>();
 
 		for (SpoonFile f : getFiles()) {
 			// we take care not to add a file that was already found in a folder
 			if (!result.contains(f)) {
 				result.add(f);
 			}
 		}
-		return result;
+
+		List<SpoonFile> resultAsList = new ArrayList<>(result);
+		return resultAsList;
```

Comment:
Actually, in order to improve the perf, it looks interesting to store the results locally in the instance to avoid computing it again each time. WDYT?

Reply:
I do not understand the code well enough unfortunately. Yesterday was my first day playing with spoon.

files appears to be mutable given addFile() and addFolder() methods. There is no need to call getFiles() here at all since getFiles() just returns a List representation of the Set<SpoonFile> files anyway. If there are thread concerns then there are already issues with non-protected blocks _so_ I think just using the files as is works ok? Submitting new change today.

Reply:
Yeah actually I had second thought about my comment after posting it. So let's forgot this idea for now.